mcp/mastra/verticals/commute/tools.ts

```diff
@@ -0,0 +1,464 @@
+      userRatingsTotal: place.user_ratings_total,
+      description: place.types?.join(', '),
+      types: place.types,
+    }));
+
+    return {
+      places,
+      routeInfo: {
+        origin: route.legs[0].start_address,
+        destination: route.legs[route.legs.length - 1].end_address,
+      },
+    };
+  },
+});
+
+export const searchPlacesByDistance = createTool({
+  id: 'searchPlacesByDistance',
+  description: 'Search for places near a location, ordered by distance (closest first)',
+  inputSchema: z.object({
+    location: z.string().describe('Center location as address or coordinates'),
+    searchQuery: z.string().describe('What to search for (e.g., "restaurant", "EV charger", "hotel")'),
+    radius: z.number().default(5000).describe('Search radius in meters (max 50000)'),
+    maxResults: z.number().default(10).describe('Maximum number of results to return (1-20)'),
+  }),
+  outputSchema: z.object({
+    places: z.array(
+      z.object({
+        name: z.string(),
+        address: z.string(),
+        location: z.object({
+          lat: z.number(),
+          lng: z.number(),
+        }),
+        rating: z.number().optional(),
+        userRatingsTotal: z.number().optional(),
+        description: z.string().optional(),
+        types: z.array(z.string()).optional(),
+        distanceFromCenter: z.number().optional().describe('Distance in meters from search center'),
+      }),
+    ),
+    searchCenter: z.object({
+      address: z.string(),
+      location: z.object({
+        lat: z.number(),
+        lng: z.number(),
+      }),
+    }),
+  }),
+  execute: async ({ location, searchQuery, radius, maxResults }) => {
+    const { client, apiKey } = getGoogleMapsClient();
+
+    const geocodeResponse = await client.geocode({
+      params: {
+        address: location,
+        key: apiKey,
+      },
+    });
+
+    if (geocodeResponse.data.status !== 'OK') {
+      throw new Error(
+        `Geocoding error: ${geocodeResponse.data.status} - ${geocodeResponse.data.error_message || 'Unknown error'}`,
+      );
+    }
+
+    if (!geocodeResponse.data.results || geocodeResponse.data.results.length === 0) {
+      throw new Error(`Could not geocode location: ${location}`);
+    }
+
+    const centerLocation = geocodeResponse.data.results[0].geometry.location;
+    const centerAddress = geocodeResponse.data.results[0].formatted_address;
+
+    const placesResponse = await client.textSearch({
+      params: {
+        query: searchQuery,
+        location: centerLocation,
+        radius: Math.min(radius, 50000),
+        key: apiKey,
+      },
+    });
+
+    if (placesResponse.data.status !== 'OK' && placesResponse.data.status !== 'ZERO_RESULTS') {
+      throw new Error(
+        `Places API error: ${placesResponse.data.status} - ${placesResponse.data.error_message || 'Unknown error'}`,
+      );
+    }
+
+    const calculateDistance = (lat1: number, lon1: number, lat2: number, lon2: number): number => {
```

Comment:
Check if this is inbuilt in Google Maps node sdk, is part of some response type, or if there exists an npm package for calculating it.